src/tests/ftest/object/object_integrity.py

```diff
@@ -133,6 +134,65 @@ def reconnect(self):
                                self.obj,
                                objtype=4)
 
+    @avocado.fail_on(DaosApiError)
+    def test_invalid_tx_commit_close(self):
+        """
+        Test ID:
+            (1)DAOS-1346: Verify commit tx bad parameter behavior.
+            (2)DAOS-1343: Verify tx_close bad parameter behavior.
+            (3)DAOS-1342: Verify tx_close through daos_api.
+        Test Description:
+            Write Avocado Test to verify commit tx and close tx
+                          bad parameter behavior.
+        :avocado: tags=negative_test,neg_tx_commit,medium,vm
+        """
+        self.d_log.info("==Writing the Single Dataset for negative test...")
+        record_index = 0
+        expected_error = "RC: -1002"
+        dkey = 0
+        akey = 0
+        indata = ("{0}".format(str(akey)[0])
+                  * self.record_length[record_index])
+        c_dkey = ctypes.create_string_buffer("dkey {0}".format(dkey))
+        c_akey = ctypes.create_string_buffer("akey {0}".format(akey))
+        c_value = ctypes.create_string_buffer(indata)
+        c_size = ctypes.c_size_t(ctypes.sizeof(c_value))
+        new_transaction = self.container.get_new_tx()
```

Comment:
This should be in some try, except block as well just incase if there is some genuine daos error to get new tx which has a bug. Otherwise if this will fail, avocado will report test status as ERROR and not FAILED.

Reply:
Script updated with self.cancel to skip the rest of test incase of failure detected here.